### PR TITLE
config: update flags for etcd 2.1

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by crawford** [comment]
I'd still like a regex string to validate the package logging levels (cc @barakmich).

File: config/etcd2.go [hunk 1]
```diff
@@ -33,12 +36,21 @@ type Etcd2 struct {
...
 	Name                     string `yaml:"name"                          env:"ETCD_NAME"`
-	PeerCAFile               string `yaml:"peer_ca_file"                  env:"ETCD_PEER_CA_FILE"`
+	PeerCAFile               string `yaml:"peer_ca_file"                  env:"ETCD_PEER_CA_FILE"                deprecated:"peer_ca_file implies peer_trusted_ca_file and peer_client_cert_auth"`
 	PeerCertFile             string `yaml:"peer_cert_file"                env:"ETCD_PEER_CERT_FILE"`
 	PeerKeyFile              string `yaml:"peer_key_file"                 env:"ETCD_PEER_KEY_FILE"`
-	Proxy                    string `yaml:"proxy"                         env:"ETCD_PROXY" valid:"^(on|off|readonly)$"`
+	PeerClientCertAuth       bool   `yaml:"peer_client_cert_auth"         env:"ETCD_PEER_CLIENT_CERT_AUTH"`
+	PeerTrustedCAFile        string `yaml:"peer_trusted_ca_file"          env:"ETCD_PEER_TRUSTED_CA_FILE"`
+	Proxy                    string `yaml:"proxy"                         env:"ETCD_PROXY"                       valid:"^(on|off|readonly)$"`
+	ProxyDialTimeout         int    `yaml:"proxy_dial_timeout"            env:"ETCD_PROXY_FAILURE_WAIT"`
```
Comment (yichengq) [inline config/etcd2.go]: wrong one.

Comment (crawford) [inline config/etcd2.go]: oops :)

File: config/etcd2.go [hunk 2]
```diff
@@ -33,12 +36,21 @@ type Etcd2 struct {
...
-	Proxy                    string `yaml:"proxy"                         env:"ETCD_PROXY" valid:"^(on|off|readonly)$"`
+	PeerClientCertAuth       bool   `yaml:"peer_client_cert_auth"         env:"ETCD_PEER_CLIENT_CERT_AUTH"`
+	PeerTrustedCAFile        string `yaml:"peer_trusted_ca_file"          env:"ETCD_PEER_TRUSTED_CA_FILE"`
+	Proxy                    string `yaml:"proxy"                         env:"ETCD_PROXY"                       valid:"^(on|off|readonly)$"`
+	ProxyDialTimeout         int    `yaml:"proxy_dial_timeout"            env:"ETCD_PROXY_DIAL_TIMEOUT"`
+	ProxyFailureWait         int    `yaml:"proxy_failure_wait"            env:"ETCD_PROXY_FAILURE_WAIT"`
+	ProxyReadTimeout         int    `yaml:"proxy_read_timeout"            env:"ETCD_PROXY_READ_TIMEOUT"`
+	ProxyRefreshInterval     int    `yaml:"proxy_refresh_interval"        env:"ETCD_PROXY_REFRESH_INTERVAL"`
+	ProxyWriteTimeout        int    `yaml:"proxy_write_timeout"           env:"ETCD_PROXY_WRITE_TIMEOUT"`
 	SnapshotCount            int    `yaml:"snapshot_count"                env:"ETCD_SNAPSHOTCOUNT"`
```
Comment (yichengq) [inline config/etcd2.go]: this one is wrong.. SNAPSHOT_COUNT

Comment (crawford) [inline config/etcd2.go]: fixed

**Comment by yichengq** [comment]
lgtm

**Pull request merged**